components/gitpod-cli/cmd/rebuild.go

```diff
@@ -0,0 +1,232 @@
+			Command: cmd.Name(),
+		})
+		defer event.Send(ctx)
+
+		gitpodConfig, err := utils.ParseGitpodConfig(wsInfo.CheckoutLocation)
+		if err != nil {
+			fmt.Println("The .gitpod.yml file cannot be parsed: please check the file and try again")
+			fmt.Println("")
+			fmt.Println("For help check out the reference page:")
+			fmt.Println("https://www.gitpod.io/docs/references/gitpod-yml#gitpodyml")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MalformedGitpodYaml)
+			return
+		}
+
+		if gitpodConfig == nil {
+			fmt.Println("To test the image build, you need to configure your project with a .gitpod.yml file")
+			fmt.Println("")
+			fmt.Println("For a quick start, try running:\n$ gp init -i")
+			fmt.Println("")
+			fmt.Println("Alternatively, check out the following docs for getting started configuring your project")
+			fmt.Println("https://www.gitpod.io/docs/configure#configure-gitpod")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MissingGitpodYaml)
+			return
+		}
+
+		var baseimage string
+		switch img := gitpodConfig.Image.(type) {
+		case nil:
+			baseimage = ""
+		case string:
+			baseimage = "FROM " + img
+		case map[interface{}]interface{}:
+			dockerfilePath := filepath.Join(wsInfo.CheckoutLocation, img["file"].(string))
+
+			if _, err := os.Stat(dockerfilePath); os.IsNotExist(err) {
+				fmt.Println("Your .gitpod.yml points to a Dockerfile that doesn't exist: " + dockerfilePath)
+				event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileNotFound).Send(ctx)
+				return
+			}
+			dockerfile, err := os.ReadFile(dockerfilePath)
+			if err != nil {
+				log.Fatal("Could not read the Dockerfile")
+				return
+			}
+			if string(dockerfile) == "" {
+				fmt.Println("Your Gitpod's Dockerfile is empty")
+				fmt.Println("")
+				fmt.Println("To learn how to customize your workspace, check out the following docs:")
+				fmt.Println("https://www.gitpod.io/docs/configure/workspaces/workspace-image#use-a-custom-dockerfile")
+				fmt.Println("")
+				fmt.Println("Once you configure your Dockerfile, re-run this command to validate your changes")
+				event.Set("ErrorCode", utils.RebuildErrorCode_DockerfileEmpty)
+				return
+			}
+			baseimage = "\n" + string(dockerfile) + "\n"
+		default:
+			fmt.Println("Check your .gitpod.yml and make sure the image property is configured correctly")
+			event.Set("ErrorCode", utils.RebuildErrorCode_MalformedGitpodYaml)
+			return
+		}
+
+		if baseimage == "" {
+			// TODO: should this fallback to workspace full?
+			fmt.Println("Your project is not using any custom Docker image.")
+			fmt.Println("Check out the following docs, to know how to get started")
+			fmt.Println("")
+			fmt.Println("https://www.gitpod.io/docs/configure/workspaces/workspace-image#use-a-public-docker-image")
+			return
+		}
+
+		err = os.WriteFile(filepath.Join(tmpDir, "Dockerfile"), []byte(baseimage), 0644)
+		if err != nil {
+			fmt.Println("Could not write the temporary Dockerfile")
```

Comment:
TODO: Add `event.Set("ErrorCode", ...)` here.

Reply:
I agree, but due to `defer event.Send(ctx)` up there, the `gp_command` is being recorded in Segment anyway.

So if we're not going to log that there was an error, we shouldn't log that the command was executed either.

I mean, we have two options:
- Always send Segment event, which will always contain one of these two fields: "errorCode" or "imageBuildDurationSeconds".
- Prevent `event.Send(ctx)` to run, in those cases considered workspace errors.

(I see you already added `event.Set("ErrorCode", ...)` for this case, but I just wanted to answer your question. I'm marking this comment as resolved now.)